### PR TITLE
Storybook: Move Menu Out of DropDown

### PR DESCRIPTION
**Pull request by BrittanyIRL**
## Summary
En route to building a typeahead and popover menu, extracting menu from dropDown to be reused. 

## Relevant Technical Choices

- Moving sampleData out of dropDown and into both testUtils and storybookUtils so that it can change if we need specific test cases without impacting storybook. 
- Adding a storybook section for just menu.
- Renaming `selectButtonId` prop in menu to `parentId`

## User-facing changes

None

## Testing Instructions

Verify that the dropDown in storybook/DesignSystem still works as expected
See that there's now a storybook/DesignSystem/components/Menu that's just the menu.

---

Fixes #

### CODE REVIEW

File: assets/src/design-system/components/menu/menu.js [hunk 1]
```diff
@@ -23,14 +23,31 @@ import PropTypes from 'prop-types';
...
+ * @param {Function} props.onMenuItemClick Triggered when a user clicks or presses 'Enter' on an option.
+ * @param {Function} props.onDismissMenu Triggered when a user escapes a menu or clicks outside of it.
+ * @param {Function} props.renderItem If present when menu is open, will override the base list items rendered for each option, the entire item and whether it is selected will be returned and allow you to style list items internal to a list item without affecting dropdown functionality.
+ * @param {string} props.activeValue the selected value of the dropDown. Should correspond to a value in the options array of objects.
+ * @param {string} props.menuAriaLabel Specific label to use as menu's aria label for screen readers.
+ * @param {string} props.parentId if in a dropDownMenu, this is the id associated with the button that controls when the menu is visible.
+ *
+ */
+
+const Menu = ({
```
Comment (carloskelly13) [inline assets/src/design-system/components/menu/menu.js:50]: I'm not sure we need JSDocs for props if we have `propTypes`. I think JSDocs should be used for functions or utils.

File: assets/src/design-system/testUtils/sampleData.js [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    options: [
+      { value: 'dog-1', label: 'Snoopy' },
+      { value: 'dog-2', label: 'Scooby' },
+    ],
+  },
+  {
+    label: 'tricky content',
+    options: [
+      { value: 0, label: '0 as a number' },
+      { value: false, label: 'false as a boolean' },
```
Comment (carloskelly13) [inline assets/src/design-system/testUtils/sampleData.js]: Screen Shot 2021-01-04 at 9 00 19 AM

The problem with these are you cannot select falsy values. Now this is just a story so I'm not sure we need to address this, but ideally we'd never have an option with a falsy value because it gets ignored on select. I'd say we either drop this example or wrap the falsy values in a string.

File: assets/src/design-system/components/menu/stories/index.js [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+  reallyLongOptions,
+} from '../../../storybookUtils/sampleData';
+import { getOptions } from '../utils';
+
+export default {
+  title: 'DesignSystem/Components/Menu',
+};
+
+const Container = styled.div`
+  width: ${({ narrow }) => (narrow ? 150 : 400)}px;
```
Comment (carloskelly13) [inline assets/src/design-system/components/menu/stories/index.js]: I'd love to extract out `NARROW_WIDTH` and `STANDARD_WIDTH` as consts just so it's very clear what we are setting here.

**Review by carloskelly13 — APPROVED** [review]
Just a couple suggestions, then this will be good to go.

File: assets/src/design-system/components/menu/menu.js (hunk 1 above)
Comment (BrittanyIRL) [reply assets/src/design-system/components/menu/menu.js:50]: It does seem a little like overkill, but i'm finding the explanations helpful. The existing font typeahead in the story editor has props defined in JSDocs and I've found it helpful comparing and following paths of what specific props are for. Especially with the menu hopefully getting used as a menu for dropDowns, typeaheads, and popovers, i think it doesn't hurt to have the added description.

File: assets/src/design-system/testUtils/sampleData.js (hunk 2 above)
Comment (BrittanyIRL) [reply assets/src/design-system/testUtils/sampleData.js]: I can drop the example, I'd initially done this to catch edge cases as developing the dropDown. You're right that we shouldn't have a false value in the dropDown though. 

I think I'll remove them from storybook, but set up a new test in the DropDown with this bad data to make sure it's not getting selected.

Comment (BrittanyIRL) [reply assets/src/design-system/testUtils/sampleData.js]: Made more sense to just test in the utils where sanitizing is happening, moved the "bad options" to their own example.
